File: Go/Algorithms/2000.go
package main

import "strings"

func reversePrefix(word string, ch byte) string {
	pos := strings.IndexByte(word, ch)
	if pos == -1 {
		return word
	}

	res := make([]byte, pos+1)
	for i := 0; i <= pos; i++ {
		res[pos-i] = word[i]
	}
	for i := pos + 1; i < len(word); i++ {
		res = append(res, word[i])
	}

	return string(res)
}

File: Rust/Algorithms/2000.rs
impl Solution {
    pub fn reverse_prefix(word: String, ch: char) -> String {
        if let Some(idx) = word.find(ch) {
            let (start, end) = word.split_at(idx + 1);
            let mut res: String = start.chars().rev().collect();
            res.push_str(end);
            res
        } else {
            word
        }
    }
}